rocAL/rocAL/source/api/rocal_api_augmentation.cpp

```diff
@@ -61,425 +83,416 @@ rocalSequenceRearrange(RocalContext p_context,
     }
     return output;
 }
-/*
-RocalImage  ROCAL_API_CALL
+
+RocalTensor  ROCAL_API_CALL
 rocalRotate(
         RocalContext p_context,
-        RocalImage p_input,
+        RocalTensor p_input,
         bool is_output,
         RocalFloatParam p_angle,
         unsigned dest_width,
-        unsigned dest_height)
-{
-
-    Image* output = nullptr;
+        unsigned dest_height,
+        RocalResizeInterpolationType interpolation_type,
+        RocalTensorLayout output_layout,
+        RocalTensorOutputType output_datatype) {
+    Tensor* output = nullptr;
     if ((p_context == nullptr) || (p_input == nullptr)) {
         ERR("Invalid ROCAL context or invalid input image")
         return output;
     }
     auto context = static_cast<Context*>(p_context);
-    auto input = static_cast<Image*>(p_input);
+    auto input = static_cast<Tensor*>(p_input);
     auto angle = static_cast<FloatParam*>(p_angle);
-    try
-    {
-        if(dest_width == 0 || dest_height == 0)
-        {
-            dest_width = input->info().width();
-            dest_height = input->info().height_single();
+    try {
+        if(dest_width == 0 || dest_height == 0) {
+            dest_width = input->info().max_shape()[0];
+            dest_height = input->info().max_shape()[1];
         }
-        // For the rotate node, user can create an image with a different width and height
-        ImageInfo output_info = input->info();
-        output_info.width(dest_width);
-        output_info.height(dest_height);
-
-        output = context->master_graph->create_image(output_info, is_output);
-
-        // If the user has provided the output size the dimension of all the images after this node will be fixed and equal to that size
-        if(dest_width != 0 && dest_height != 0)
-            output->reset_image_roi();
-        std::shared_ptr<RotateNode> rotate_node =  context->master_graph->add_node<RotateNode>({input}, {output});
-        rotate_node->init(angle);
+        RocalTensorlayout op_tensor_layout = static_cast<RocalTensorlayout>(output_layout);
+        RocalTensorDataType op_tensor_datatype = static_cast<RocalTensorDataType>(output_datatype);
+        TensorInfo output_info = input->info();
+        output_info.set_tensor_layout(op_tensor_layout);
+        output_info.set_data_type(op_tensor_datatype);
+        
+        // For the rotate node, user can create a tensor with a different width and height
+        std::vector<size_t> out_dims = output_info.dims();
+        modify_dims_width_and_height(output_info.layout(), out_dims, dest_width, dest_height);
```

Comment:
we don't want to have this function defined in this API class. Probably you need to add another fun called calculate_dims_xx() in the tensor class which claculates dimensions based on width, height, layout etc. API definitions whould only have calls to add augmentations. Here I see lot of calls to ceate tensor and add roi

Reply:
Done